appservice/src/editScmType.ts

```diff
@@ -39,24 +40,24 @@ export async function editScmType(client: SiteClient, node: AzureTreeItem): Prom
     return newScmType;
 }
 
-async function showScmPrompt(currentScmType: string): Promise<string> {
+async function showScmPrompt(currentScmType: string): Promise<ScmType> {
     const currentSource: string = localize('currentSource', '(Current source)');
-    const scmQuickPicks: IAzureQuickPickItem<string | undefined>[] = [];
+    const scmQuickPicks: IAzureQuickPickItem<ScmType | undefined>[] = [];
     // generate quickPicks to not include current type
     for (const scmType of Object.keys(ScmType)) {
         if (scmType === currentScmType) {
             // put the current source at the top of the list
             scmQuickPicks.unshift({ label: scmType, description: currentSource, data: undefined });
         } else {
-            scmQuickPicks.push({ label: scmType, description: '', data: scmType });
+            scmQuickPicks.push({ label: scmType, description: '', data: <ScmType>scmType });
```

Comment:
Honestly there's a bug here and you're just getting lucky because the key and value are the same:
```
export enum ScmType {
    None = 'None', // default scmType
    LocalGit = 'LocalGit',
    GitHub = 'GitHub'
}
```

If your enum was like this it wouldn't work:
```
export enum ScmType {
    None = 'hi', // default scmType
    LocalGit = 'nathan',
    GitHub = 'wassup'
}
```

You have to do something like this:
```
for (const key of Object.keys(ScmType)) {
    const scmType = <ScmType>ScmType[key];
    ...
}
```